nilearn/decomposition/single_pca.py

```diff
@@ -0,0 +1,320 @@
+"""
+PCA dimension reduction on single subjects
+"""
+import warnings
+
+from scipy import linalg
+import nibabel
+from sklearn.base import BaseEstimator, clone
+from sklearn.externals.joblib import Parallel, delayed, Memory
+from sklearn.utils.extmath import randomized_svd
+from sklearn.utils.validation import check_random_state
+import numpy as np
```

Comment:
numpy import should be just before scipy import: we do from more general to less general (ie more domain-specific)